### PR TITLE
Enable eslint-plugin-unicorn recommended rules in `base` config

### PR DESCRIPTION
**Pull request by bmish**
This plugin has a large variety of amazing lint rules, especially for adopting new JavaScript syntax/functions, and most of them are autofixable. I've really enjoyed using them in some of my applications already.

Note that we are already using this plugin internally ourselves.

https://github.com/sindresorhus/eslint-plugin-unicorn

### CODE REVIEW

**Comment by ghaagsma** [comment]
Does the label `breaking` invoke a different semantic versioning bump for this change?

**Comment by bmish** [comment]
Yes, I'll include this in a major version bump soon.

File: lib/config/base.js [hunk 1]
```diff
@@ -6,12 +6,13 @@ module.exports = {
   extends: [
     'eslint:recommended',
     'plugin:eslint-comments/recommended',
+    'plugin:unicorn/recommended',
     'prettier',
```
Comment (maxbeatty) [inline lib/config/base.js:10]: should we also add `prettier/unicorn`? https://github.com/prettier/eslint-config-prettier/blob/master/unicorn.js

Comment (bmish) [reply lib/config/base.js:10]: Nice find, I'll add that to this PR. I'm still planning to do some more testing of all these changes before moving forward with them.

**Pull request merged**